Yet Another Fugitives PR: Bounty Hunters, Features, Fixes, Treason, Removals!

tralezab opened the pull request:
## what this does in full:

- Waldo is less common. He's kind of immersion breaking.
- Waldo is actually waldo now. Whoops
- Yalp Elor is removed. Veeeery broken implementation.
- Removed the "armed" part of the hunter's arrival announcement. It riled security up and so this will hopefully encourage more interesting interactions
- The russian ship has some big changes. No more ripley, everyone gets a space suit, and less welding tanks. I should really redo this whole thing (and I might still) but this at least fixes the biggest problems with the ship.
- Russians now get a mosin, and have a 50/50 to spawn with either the bear hat or a ushanka.
- Hunters and fugitives get their own hud:
![image](https://user-images.githubusercontent.com/40974010/60061346-ac8be700-96a9-11e9-84f7-baf230124e43.png)
- Space cops get handcuffs instead of an extra mag.
- Bounty Hunters are a new hunter! They get some exotic tools to catch their prey.
Pictures soon!

fixes #44250

## Changelog
:cl: traelezab with Dunham and Fikou for sprites, thanks fellas
add: There is a new fugitive hunter, the bounty hunters! They're barely better than space pirates!
add: The russian shuttle has undergone some changes, namely the removal of the ripley, and the addition of space suits.
del: The cultists have lost their god, RIP yalp elor.
tweak: Other various improvements like different huds for the hunters and fugitives, a less threatening announcement message, and more.
/:cl:

tralezab committed:
mostly fixes in this commit

tralezab committed:
removes old_god from the dme

tralezab committed:
russian shuttle improvements

MrDoomBringer commented:
samus bounty hunter when

habader commented:
rip yelp

also can you make the fugitives actual antags? trying to collect enough gear to survive before the hunters come is actually really difficult when you've got to weave and duck around bullshit semi-antag rules

tralezab committed:
TODO: fix the hook gun

tralezab committed:
Merge branch 'master' into fugitive-features-fixes-etc

tralezab committed:
shotgun finished, next commit shuttle

ATH1909 commented:
Will Yalp Elor ever return?

Can't you just copy some code from Dark Spirits for Yalp Elor? Make him a subtype of them, perhaps?

Alternatively, maybe you could make Yalp Elor just be an imaginary friend of the lead cultist?

fludd12 commented:
If there isn't already a plan for it, I might do something regarding the cultists' god in the future - of course, with how much is already on my plate, I won't make any promises...

tralezab commented:
once i finish the map (and squash this conflict) we'll be ready to go

tralezab committed:
bounty hunting shuttle

SpaceManiac commented:
merge conflict, it's all in icons so please let me know if the script cannot handle it

tralezab commented:
will fix an american proportion of merge conflicts on the most american day of the year, hold on

tralezab committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into fugitive-features-fixes-etc

tralezab committed:
back up to speed on the dmis

tralezab commented:
hol up

SpaceManiac commented:
sorry about that, conflict caused by the #44962 merge

tralezab commented:
all good, i'll get through this real quick

tralezab committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into fugitive-features-fixes-etc

SpaceManiac commented:
@tralezab might be too late if you're already deep in it, but I finally wrote up the wiki page for our icon merger script https://github.com/tgstation/tgstation/wiki/Resolving-icon-conflicts

tralezab commented:
i just finished with the conflicts, at least it'll be useful for a project in the near future. I just noticed some last minute problems I should fix, then this'll be good to go.

tralezab committed:
readds poster 46 and adds the shuttle finally

tralezab commented:
seems good to go 👍

Fikou commented:
yee haw

tralezab committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into fugitive-features-fixes-etc

tralezab commented:
more merge conflict handling, hold on

tralezab committed:
dmi fixes

tralezab commented:
and once again this is ready

AnturK commented on _maps/templates/shelter_2.dmm:
```diff
@@ -8,6 +8,17 @@
 	},
 /turf/closed/wall/mineral/titanium/survival/pod,
 /area/survivalpod)
+"c" = (
```
What's going on here ?

tralezab replied:
I changed the curtains icons since their icons are different now

AnturK commented on code/modules/antagonists/fugitive/fugitive_outfits.dm:
```diff
@@ -81,3 +83,76 @@
+	if(visualsOnly)
+		return
+	H.set_species(/datum/species/synth)
+	H.set_species(/datum/species/synth)//setting twice removes the skin
```
This sounds off.

optimumtact replied:
it's definitely off

tralezab replied:
What's the other way to do this?

optimumtact replied:
Fix whatever bug is causing their sprite to not update.

I dont' think working around it is a good idea, just set it once and leave it broken for someone to fix.

SpaceManiac replied:
*Is* this a bug? For some reason I thought that changing species to Synth is intended to preserve your old appearance.

optimumtact replied:
You might be right, I do believe that is the lore of synths so it does sound appropriate.

Probably there is a var that can be passed through when changing the species, or a proc that can be called. I'll dig in and see

tralezab replied:
i found a better way to do this through a proc the synth species has but i need to mess around with it a bit more.

AnturK commented on code/modules/mining/lavaland/necropolis_chests.dm:
```diff
@@ -442,6 +442,20 @@
+	name = "hook"
+	ammo_type = /obj/item/ammo_casing/magic/hook/bounty
+
+/obj/item/gun/magic/hook/bounty/shoot_with_empty_chamber(mob/living/user as mob|obj)
```
This is not a verb, remove `as mob|obj`

tralezab replied:
👌 done

AnturK commented on code/modules/projectiles/guns/ballistic/shotgun.dm:
```diff
@@ -210,3 +209,46 @@
+	hook = new /obj/item/gun/magic/hook/bounty(src)
+
+/obj/item/gun/ballistic/shotgun/doublebarrel/hook/AltClick(mob/user)
+	if(toggled)
```
No adjacency or ghost checks.

tralezab replied:
added

tralezab commented:
alright lets finish this

tralezab committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into fugitive-features-fixes-etc

tralezab committed:
traps need fixing, work out whatever is happening with synths

tralezab committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into fugitive-features-fixes-etc

tralezab committed:
super final touches, see desc

map ports + proper spawners, pinpointer and traps functional. other misc changes

tralezab committed:
removes verb stuff from non verb stuff

tralezab commented:
review handled, shuttle fixed (how many times have i said that?) and supposedly done. I might add in one more costume if this stays open awhile since i can't mess that up too bad.

AnturK commented on code/game/objects/structures/traps.dm:
```diff
@@ -81,10 +85,79 @@
-	L.Paralyze(100)
+	L.Paralyze(stun_time)
+
+/obj/structure/trap/stun/hunter
```
Don't these things have anitmagic checks and such wizard stuff.

tralezab replied:
done

AnturK commented on code/game/objects/structures/traps.dm:
```diff
@@ -81,10 +85,79 @@
+	to_chat(user, "<span class=notice>You set up [src]. Examine while close to disarm it.</span>")
+	var/obj/structure/trap/stun/hunter/trap = new(get_turf(src))
+	trap.name = name
+	qdel(src)
```
Why not stuff this in the trap, instead of remaking it every time ?

tralezab replied:
there are quirks with cooldown, but i can do that.

AnturK reviewed (commented):
Looks mostly alright now

nemvar commented:
Fixes #44250 should this get merged. Well... in a sense at least.

tralezab commented:
I'll add that to the OP!

tralezab committed:
review (see desc)

antimagic checks and it now the two objects move inside each other instead

tralezab committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into fugitive-features-fixes-etc

tralezab committed:
dmi conflicts

optimumtact commented on code/game/objects/structures/traps.dm:
```diff
@@ -81,10 +86,84 @@
+	radio.talk_into(src, "Fugitive has triggered this trap in the [get_area_name(src)]!", RADIO_CHANNEL_COMMON)
+
+/obj/item/bountytrap/attack_self(mob/living/user)
+	if(!user || !user.temporarilyRemoveItemFromInventory(src))//visibly unequips
```
why not just https://github.com/tgstation/tgstation/blob/412c3b63e39c60ecd1dfdfa245e040e92b4ff59b/code/modules/mob/inventory.dm#L300

tralezab replied:
sure

optimumtact commented:
![](https://file.house/Fu6H.png)

tralezab commented:
what is not supposed to exist?

tralezab committed:
PLUNGERS. BEGONE.

tralezab commented:
handled

optimumtact commented:
I actually forgot to leave a comment but I wasn't sure why you were changing the WC sprites

tralezab commented:
because they were hard coded to have a single sprite and i was adding themed curtains for the shuttle, i needed to rename them
